tools/java/org/unicode/cldr/test/ExampleGenerator.java

```diff
@@ -167,44 +178,157 @@ public CLDRFile getCldrFile() {
     private CLDRFile englishFile;
     Matcher URLMatcher = URL_PATTERN.matcher("");
 
-    /*
+    /**
      * The cache is accessed only by getExampleHtml and updateCache.
-     * Its key is built from an xpath, a value for that xpath, and a letter to indicate the ExampleType (ENGLISH or NATIVE).
+     * Its key is built from an xpath, and a value for that xpath.
      * Its value is an HTML string showing example(s) using that value for that path, for the locale of this ExampleGenerator.
      *
      * Note that this cache is internal to each ExampleGenerator. Compare TestCache.exampleGeneratorCache,
      * which is at a higher level, caching entire ExampleGenerator objects, one for each locale.
-     *
-     * TODO: simplify/clarify the ENGLISH/NATIVE distinction, to facilitate performance improvements.
-     * The general-purpose mechanism that currently appears to support both ENGLISH and NATIVE for each
-     * ExampleGenerator doesn't currently appear to be used or needed. If the two types are not really mixed
-     * in the same cache (or even in same ExampleGenerator), then E/N needn't be part of cache key (or
-     * require separate caches). Also, at least within Survey Tool, englishFile == cldrFile if (and only
-     * if) the ExampleGenerator is used for ExampleType.ENGLISH. In Survey Tool, ExampleType.ENGLISH is
-     * only actually used for a single ExampleGenerator, namely sm.getTranslationHintsExample().
-     *
-     * Reference: https://unicode-org.atlassian.net/browse/CLDR-12020
      */
     private Map<String, String> cache = new ConcurrentHashMap<String, String>();
 
     /**
-     * For this (locale-specific) ExampleGenerator, clear the cached value for the given xpath,
-     * and for any xpaths whose examples might also depend on this xpath.
+     * AVOID_CLEARING_CACHE: work in progress, keep false until it becomes beneficial and reliable.
+     * Reference: https://unicode-org.atlassian.net/browse/CLDR-13331
+     */
+    private static final boolean AVOID_CLEARING_CACHE = false;
```

Comment:
should it be an env var?

Reply:
Once it's working reliably and demonstrated to improve performance, we might not need the boolean anymore, we might just make it effectively true permanently. Though, I suppose we could keep it in case later on we want to double-check whether it's still improving performance, and/or still working correctly by producing the same examples as it would if false. I don't really see a use case for making it an environment variable.